### DISCUSSION TITLE
Why extract multiple zero_point / scale values from the same quantization group?

### DISCUSSION THREAD

**Discussion by frankxyy** [turboderp/exllamav2#214] [General]
Hi, I read the code of gptq reconstruct and find this line:
<img width="871" alt="image" src="https://github.com/turboderp/exllamav2/assets/6971044/6465624e-ccbc-47d8-a14f-e7c6b239e8db">

It seems that 4 values of zero_point are read out from the matrix for the same group. From the theory of gptq quantization, isn't it that there is only one particular zero_point and scale value for one specific group?

**Comment by turboderp** [comment]
It does four columns in each thread, and each column has its own scale and offset. It's faster this way, loading four offsets as one `uint16`, four scales as two `half2`s and 4 x 8 weights an `int4`.

**Reply by frankxyy** [reply]
Got it! Thanks a lot
